### PR TITLE
data/find: refactored code

### PR DESCRIPTION
**Pull request by azizk**
* Simplified toggle_case_sensitive(), next_occurrence() and is_matching_whole_words().
* Avoid unnecessary cloning in next_occurrence().
* Improved doc comments.
* Removed obsolete code.
* Inserted spaces for better readability.
* Moved is_matching_whole_words() outside to module level.

### CODE REVIEW

**Review by panekj — CHANGES_REQUESTED** [review]

**Comment by azizk** [comment]
Thanks for the quick review :+1:

File: lapce-data/src/find.rs [hunk 1]
```diff
@@ -278,48 +267,50 @@ impl Find {
...
         end: usize,
         include_slop: bool,
     ) {
-        if self.search_string.is_none() {
+        let search_string = if let Some(s) = self.search_string.as_ref() {
+            s
+        } else {
             return;
-        }
+        };
```
Comment (panekj) [inline lapce-data/src/find.rs:305]:
```rs
        let Some(search_string) = self.search_string.as_ref() else {
            return;
        };
```

Comment (azizk) [reply lapce-data/src/find.rs:305]: I'm confused a bit. Is that code syntactically correct? Is `x else { return; }` possible?

Comment (panekj) [reply lapce-data/src/find.rs:305]: https://doc.rust-lang.org/stable/rust-by-example/flow_control/let_else.html

Comment (azizk) [reply lapce-data/src/find.rs:305]: Oh! It's been available since `rust 1.65` :smile:

**Comment by panekj** [comment]
`lapce-data` isn't used anymore, all code is in `lapce-app`

**Comment by Nyabinary** [comment]
Is this stalled?

**Comment by azizk** [comment]
@panekj But the source files are still there and weren't deleted? Does this mean `lapce-data` is completely obsolete and `lapce-app` has its own finder functions?

@Nyabinary I don't know. I'm not sure what to do atm.

**Comment by panekj** [comment]
> Does this mean `lapce-data` is completely obsolete and `lapce-app` has its own finder functions?

Yes, old files are kept for reference until ported

**Comment by azizk** [comment]
I'm not up-to-date anymore. Has this been ported yet and can you point me to the location? If not, why not merge this in?

**Comment by Nyabinary** [comment]
@azizk I don't believe the project has been ported yet.

**Comment by azizk** [comment]
Okay, so in that case this PR should be merged then?

**Comment by azizk** [comment]
The whole directory has been finally removed: 774008f54bc5843afa8cbf8fa766b620c545aa27

The new file is located in `lapce-app/src/find.rs` now, but it's been reworked quite a bit so that my changes don't seem to be useful anymore... :disappointed:

**Pull request closed without merging**